src/components/structures/IndicatorScrollbar.js

```diff
@@ -131,17 +131,28 @@ export default class IndicatorScrollbar extends React.Component {
             // the harshness of the scroll behaviour. Should be a value between 0 and 1.
             const yRetention = 1.0;
 
-            // Check for trackpad users every so often to avoid boosting their scroll.
-            // See https://github.com/vector-im/riot-web/issues/10005
+            // whenever we see horizontal scrolling, assume the user is on a trackpad
+            // for at least the next 1 minute.
             const now = new Date().getTime();
-            if (now >= this._checkAgainForTrackpad) {
-                this._likelyTrackpadUser = Math.abs(e.deltaX) > 0;
-                this._checkAgainForTrackpad = now + (15 * 60 * 1000); // 15min
+            if (Math.abs(e.deltaX) > 0) {
+                this._likelyTrackpadUser = true;
+                this._checkAgainForTrackpad = now + (1 * 60 * 1000);
+            } else {
+                // if we haven't seen any horizontal scrolling for a while, assume
+                // the user might have plugged in a mousewheel
+                if (this._likelyTrackpadUser && now >= this._checkAgainForTrackpad) {
+                    this._likelyTrackpadUser = false;
+                }
+            }
+
+            // don't mess with the horizontal scroll for trackpad users
+            // See https://github.com/vector-im/riot-web/issues/10005
+            if (this._likelyTrackpadUser) {
+                return;
             }
 
-            const safeToBoost = !this._likelyTrackpadUser;
+            if (Math.abs(e.deltaX) <= xyThreshold) { // we are vertically scrolling.
 
```

Comment:
the linter hates this line for some dumb reason. It wants there to be no blank space.

Reply:
our linter is totally unhelpfully overzealous >:(

Reply:
Fixed linter in https://github.com/matrix-org/matrix-react-sdk/pull/3148/commits/00dfdfe7f171fad3822efde3a9bdf2851f1cd910